src/client/pluginmanager.js

```
@@ -95,22 +113,66 @@ define([
+            pluginManager.initializePlugin(pluginIdOrClass)
+                .then(function (plugin_) {
+                    plugin = plugin_;
+                    
+                    pluginEntry = {
+                        name: plugin.getName(),
+                        plugin: plugin,
+                        metadata: plugin.pluginMetadata,
+                        context: context,
+                        canBeAborted: plugin.pluginMetadata.canBeAborted,
+                        start: Date.now(),
+                        clientSide: true,
+                        executionId: null
+                    };
+
+                    executionId = generateKey({name: pluginEntry.name, start: pluginEntry.start}, gmeConfig);
+                    pluginEntry.executionId = executionId;
+                    runningPlugins[executionId] = pluginEntry;
+                    
+                    return pluginManager.configurePlugin(plugin, context.pluginConfig, context.managerConfig);
+                })
+                .then(function () {
+                    return Q.ninvoke(pluginManager, 'runPluginMain', plugin);
+                })
+                .then(function (result) {
+                    delete runningPlugins[executionId];
+                    client.dispatchEvent(client.CONSTANTS.PLUGIN_FINISHED, pluginEntry);
+                    callback(null, result);
+                })
+                .catch(function (err) {
+                    delete runningPlugins[executionId];
+                    client.dispatchEvent(client.CONSTANTS.PLUGIN_FINISHED, pluginEntry);
+                    var pluginResult = pluginManager.getPluginErrorResult(
+                        plugin.getId(),
+                        plugin.getName(),
+                        'Exception was raised, err: ' + err.stack, plugin && plugin.projectId
+                    );
+                    logger.error(err.stack);
+                    callback(err.message, pluginResult);
+                });
```

`.done()` otherwise exceptions can be swallowed here